app/src/test/java/uk/ac/ebi/atlas/experimentpage/json/JsonBaselineExperimentControllerWIT.java

```diff
@@ -0,0 +1,81 @@
+package uk.ac.ebi.atlas.experimentpage.json;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import uk.ac.ebi.atlas.configuration.TestConfig;
+import uk.ac.ebi.atlas.testutils.JdbcUtils;
+
+import javax.inject.Inject;
+import javax.sql.DataSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+@WebAppConfiguration
+@ContextConfiguration(classes = TestConfig.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class JsonBaselineExperimentControllerWIT {
+    @Inject
+    private DataSource dataSource;
+
+    @Inject
+    private JdbcUtils jdbcUtils;
+
+    @Autowired
+    private WebApplicationContext wac;
+
+    private MockMvc mockMvc;
+
+    @BeforeAll
+    void populateDatabaseTables() {
+        ResourceDatabasePopulator populator = new ResourceDatabasePopulator();
+        populator.addScripts(new ClassPathResource("fixtures/experiment-fixture.sql"));
+        populator.execute(dataSource);
+    }
+
+    @AfterAll
+    void cleanDatabaseTables() {
+        ResourceDatabasePopulator populator = new ResourceDatabasePopulator();
+        populator.addScripts(new ClassPathResource("fixtures/experiment-delete.sql"));
+        populator.execute(dataSource);
+    }
+
+    @BeforeEach
+    void setUp() {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(this.wac).build();
+    }
+
+    @Test
+    void getsBaselineProteomicsDiaSwathExperimentData() throws Exception {
+        var sb = new StringBuilder();
+        this.mockMvc.perform(get("/json/experiments/{experimentAccession}", "E-PROT-28")
+                .param("type", "PROTEOMICS_BASELINE_DIA_SWATH"))
```

Comment:
I guess that this test should now fail. If it’s passing there’s something wrong, because we updated the type to be `PROTEOMICS_BASELINE_DIA`, right?

Reply:
Fixed it and tested this test case. Working fine with `PROTEOMICS_BASELINE_DIA`.